pkg/service/disable_device_test.go

```diff
@@ -0,0 +1,179 @@
+package service
+
+import (
+	"context"
+	"encoding/json"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/go-kit/kit/transport/http/jsonrpc"
+	"github.com/kolide/launcher/ee/agent/types/mocks"
+	"github.com/kolide/launcher/pkg/log/multislogger"
+	"github.com/kolide/launcher/pkg/pb/launcher"
+	"github.com/osquery/osquery-go/plugin/logger"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+func TestDeviceDisabled(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name string
+		f    func(client KolideService) error
+	}{
+		{
+			name: "RequestEnrollment",
+			f: func(client KolideService) error {
+				_, _, err := client.RequestEnrollment(context.TODO(), "enroll_secret", "host_identifier", EnrollmentDetails{})
+				return err
+			},
+		},
+		{
+			name: "RequestConfig",
+			f: func(client KolideService) error {
+				_, _, err := client.RequestConfig(context.TODO(), "node_key")
+				return err
+			},
+		},
+		{
+			name: "PublishLogs",
+			f: func(client KolideService) error {
+				_, _, _, err := client.PublishLogs(context.Background(), "node_key", logger.LogTypeStatus, nil)
+				return err
+			},
+		},
+		{
+			name: "RequestQueries",
+			f: func(client KolideService) error {
+				_, _, err := client.RequestQueries(context.Background(), "node_key")
+				return err
+			},
+		},
+		{
+			name: "PublishResults",
+			f: func(client KolideService) error {
+				_, _, _, err := client.PublishResults(context.Background(), "node_key", nil)
+				return err
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			// set up JSON-RPC test server
+			jsonRpcServerResult := `{"disable_device": false}`
```

Comment:
I think the service package became a little frankensteiny over the years. Mixing jsonRPC and gRPC implementations. I'm tempted to split them up and perhaps have a common `types` package. Not sure if it's worth the effort at this time since we're not really using the grpc portion.

Reply:
I think go-kit has a recommendation for how to do this. It's core design is all about splitting up transport implementation from protocol definition.